netket/sampler/autoreg.py

```diff
@@ -77,7 +74,9 @@ def _init_cache(sampler, model, σ, key):
 
     def _init_state(sampler, model, params, key):
         new_key, key = jax.random.split(key)
-        σ = jnp.zeros((sampler.n_chains, sampler.hilbert.size), dtype=sampler.dtype)
+        σ = jnp.zeros(
+            (sampler.n_chains_per_rank, sampler.hilbert.size), dtype=sampler.dtype
```

Comment:
don't we have a field ```.sample_shape``` or similar in the sampler? I think it would be best just to initialize sigma using that field instead of ```n_chains_per_rank``` etc, since that is more error prone...

Reply:
Actually I'm already considering to remove `σ` and `cache` in `ARDirectSamplerState` in the PR of fast autoregressive sampling, because it turns out that they're not used

Reply:
you can keep them maybe because I guess in a more general implementation these might be needed... however I am just suggesting to get the shape of sigma from the method ```sampler.sample_shape```  which in my opinion should be the only one publicly exposed since ```n_chains_per_rank``` etc is a detail of markov chain samplers, not of generic ones

Reply:
I would get rid of them. A more general sampler can define its own state, and it confuses the code.

Reply:
ok then let's remove this in the next revision